prow/plugins/trigger/push.go

```diff
@@ -48,15 +51,11 @@ func handlePE(c Client, pe github.PushEvent) error {
 		return nil
 	}
 	for _, j := range c.Config.Postsubmits[pe.Repo.FullName] {
-		if !j.RunsAgainstBranch(pe.Branch()) {
+		if shouldRun, err := j.ShouldRun(pe.Branch(), listPushEventChanges(pe)); err != nil {
```

Comment:
This isn't technically the same as it didn't check that the job set `always_run: true`.